pkgs/development/libraries/vxl/default.nix

```diff
@@ -1,23 +1,52 @@
-{ lib, stdenv, fetchFromGitHub, unzip, cmake, libtiff, expat, zlib, libpng, libjpeg }:
-stdenv.mkDerivation rec {
+{ lib
+, stdenv
+, fetchFromGitHub
+, unzip
+, cmake
+, libtiff
+, expat
+, zlib
+, libpng
+, libjpeg
+}:
+
+stdenv.mkDerivation (finalAttrs: {
   pname = "vxl";
   version = "3.5.0";
 
   src = fetchFromGitHub {
     owner = "vxl";
     repo = "vxl";
-    rev = "v${version}";
+    rev = "refs/tags/v${finalAttrs.version}";
     sha256 = "sha256-4kMpIrywEZzt0JH95LHeDLrDneii0R/Uw9GsWkvED+E=";
   };
 
-  nativeBuildInputs = [ cmake unzip ];
-  buildInputs = [ libtiff expat zlib libpng libjpeg ];
+  nativeBuildInputs = [
+    cmake
+    unzip
+  ];
+  buildInputs = [
+    libtiff
+    expat
+    zlib
+    libpng
+    libjpeg
+  ];
+
+  doCheck = true;
+  # test failure on aarch64-linux; unknown reason:
+  ctestArgs = lib.optionalString stdenv.isAarch64 "-E 'vgl_test_frustum_3d'";
+  checkPhase = ''
+    runHook preCheck
+    ctest ${finalAttrs.ctestArgs}
+    runHook postCheck
+  '';
```

Comment:
You may set `-DCMAKE_CTEST_ARGUMENTS=...` in `cmakeFlags` instead.

Reply:
Thanks for the suggestion - changed.